writers/batchwriter/batchwriter.go

```diff
@@ -122,10 +122,15 @@ func (w *BatchWriter) Close(context.Context) error {
 }
 
 func (w *BatchWriter) worker(ctx context.Context, tableName string, ch <-chan *message.WriteInsert, flush <-chan chan bool) {
-	sizeBytes := int64(0)
-	resources := make([]*message.WriteInsert, 0, w.batchSize)
+	sizeBytes, sizeRows := int64(0), int64(0)
+	resources := make([]*message.WriteInsert, 0, w.batchSize) // at least we have 1 row per record
 	ticker := writers.NewTicker(w.batchTimeout)
 	defer ticker.Stop()
+
+	send := func() {
+		w.flushTable(ctx, tableName, resources)
+		resources, sizeBytes, sizeRows = resources[:0], 0, 0
```

Comment:
`clear(resources)`? (are we misusing `clear`?)

Reply:
We didn't have this in the simple batch writer, but sure, I'll add it

Reply:
5107d0b15d1a31ff9fe8d2217f85eef66ea97e5c adds clear & also exposes `capped` to be reused